packages/contracts/contracts/universal/SimpleMultiSigWallet.sol

```diff
@@ -0,0 +1,231 @@
+
+/**
+ * @custom:upgradeable
+ * @title SimpleMultiSigWallet
+ * @notice This contract implements `ISimpleMultiSigWallet`.
+ *         Allows multiple parties to agree on transactions before execution.
+ */
+abstract contract SimpleMultiSigWallet is ISimpleMultiSigWallet, ReentrancyGuardUpgradeable {
+    /**
+     * @notice The address of the governor contract. Can be updated via upgrade.
+     */
+    UpgradeGovernor public immutable GOVERNOR;
+
+    /**
+     * @notice Address of the MultiSigWallet token contract. Can be updated via upgrade.
+     */
+    IERC5805Upgradeable public immutable TOKEN;
+
+    /**
+     * @notice A mapping of transactions submitted.
+     */
+    mapping(uint256 => Types.MultiSigTransaction) public transactions;
+
+    /**
+     * @notice A mapping of confirmations.
+     */
+    mapping(uint256 => Types.MultiSigConfirmation) public confirmations;
+
+    /**
+     * @notice Spacer for backwards compatibility.
+     */
+    uint256[3] private spacer_53_0_96;
+
+    /**
+     * @notice The number of transactions submitted.
+     */
+    uint256 public transactionCount;
+
+    /**
+     * @notice Only allow the specified governance to call the functions.
+     *         This ensures that function is only executed by governance.
+     */
+    modifier onlyGovernance(address _address) {
+        require(getVotes(_address) > 0, "SimpleMultiSigWallet: onlyGovernance");
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction exists.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     */
+    modifier transactionExists(uint256 _transactionId) {
+        require(
+            transactions[_transactionId].target != address(0),
+            "SimpleMultiSigWallet: transaction does not exist"
+        );
+        _;
+    }
+
+    /**
+     * @notice Ensure that the address is not zero address.
+     *
+     * @param _address Address resource requested.
+     */
+    modifier validAddress(address _address) {
+        require(_address != address(0), "SimpleMultiSigWallet: address is not valid");
+        _;
+    }
+
+    /**
+     * @notice Internal functions. Returns the confirmation status of a transaction.
+     *
+     * @param _transactionId Transaction ID.
+     */
+    modifier confirmed(uint256 _transactionId) {
+        require(isConfirmed(_transactionId), "SimpleMultiSigWallet: not confirmed yet");
+        _;
+    }
+
+    /**
+     * @param _governor Address of the Governor contract.
+     */
+    constructor(address payable _governor) {
+        GOVERNOR = UpgradeGovernor(_governor);
+        TOKEN = IERC5805Upgradeable(address(GOVERNOR.token()));
+    }
+
+    /**
+     * @inheritdoc ISimpleMultiSigWallet
+     */
+    function submitTransaction(
+        address _target,
+        uint256 _value,
+        bytes memory _data
+    ) public onlyGovernance(msg.sender) returns (uint256) {
+        return _submitTransaction(_target, _value, _data);
```

Comment:
Why don't we add auto-confirm when submitting transaction? With this, we don't need to confirm transaction in `SecurityCouncil.sol:requestDeletion()`
```suggestion
        uint256 transactionId = _submitTransaction(_target, _value, _data);
        confirmTransaction(transactionId);
        return transactionId;
```

Reply:
I think separating auto confirm from the behavior of submit makes the intent of the function a bit clearer, and the behavior in requestValidation is cleaner.